File: algorithms/2_medium/M/04014_minimum_total_price_after_applying_discounts/sort.c
static int cmp(const void *a_ptr, const void *b_ptr) {
    const int a = *(const int*)a_ptr;
    const int b = *(const int*)b_ptr;

    return (a < b) - (a > b);
}

double minPrice(int *prices, int prices_len, int *discounts, int discounts_len) {
    const int min_len = MIN(prices_len, discounts_len);
    qsort(prices, (size_t)prices_len, sizeof(*prices), cmp);
    qsort(discounts, (size_t)discounts_len, sizeof(*discounts), cmp);
    double res = 0.0;

    for (int i = 0; i < min_len; i += 1) {
        res += (double)prices[i] * (100 - discounts[i]) / 100;
    }

    for (int i = min_len; i < prices_len; i += 1) {
        res += (double)prices[i];
    }

    return res;
}

File: algorithms/2_medium/M/04014_minimum_total_price_after_applying_discounts/sort.rs
use std::cmp::Reverse;

impl Solution {
    pub fn min_price(mut prices: Vec<i32>, mut discounts: Vec<i32>) -> f64 {
        prices.sort_unstable_by_key(|&x| Reverse(x));
        discounts.sort_unstable_by_key(|&x| Reverse(x));
        let mut discounts_iter = discounts.into_iter();
        let mut res = 0_f64;

        for p in prices {
            if let Some(d) = discounts_iter.next() {
                res += (p as f64) * (100.0_f64 - d as f64) / 100.0_f64;
            } else {
                res += p as f64;
            }
        }

        res
    }
}
